### DISCUSSION TITLE
Skin effects frame rate drastically

### DISCUSSION THREAD

**Discussion by khoibut** [ppy/osu#19756] [Q&A]
I have been messing around with Lazer for a couple of days and I found out that skins can affect your frame rate more drastic than stable
Here is a video comparing 2 skins both are considered normal skins and nowhere near high definition

First is the one with high frame rate average of above 200fps in gameplay

https://user-images.githubusercontent.com/60182480/184528298-51b56386-d2a1-4eff-b36f-c9df9fde1b16.mp4

Second is the one with low frame rate average of 100fps in gameplay

https://user-images.githubusercontent.com/60182480/184528500-5c706728-c836-4fe8-ae97-e390e39a77c8.mp4

My pc spec is quite old but ive never experienced this in stable I really wonder what causes this?

After doing some testing i replaced the hp bar of the second skin with the first skin, the second's skin frame rate improved by a lot now averaging around 180fps

Here are the two skins used in the videos
first skin: https://drive.google.com/file/d/1ypNlrBSrZEg6n8BolxkA76ukZ_B8LA_S/view?usp=sharing
second skin: https://drive.google.com/file/d/1_R8CGhS-pjeDQRPUPHzcTUlctGx1NZ7f/view?usp=sharing

**Comment by nagi-desuuu** [comment]
Have you tried using the default skins for a comparison?
Peppy has said that some skins are known to cause immense lag due to the high number of sprites or simply their resolution.

**Reply by khoibut** [reply]
I just tried using both the osu!triangle and osu!classic skins and their frame performance is relatively the same as the first skin(whitecat skin).
The resolution of the whitecat skin and the second skin(gura skin) is quite different by comparing the hitcircle.png resolution whitecat skin has a 120x120 resolution and gura skin has a 204x204 resolution. I tried the test again with High definition turned on to use the 240x240 resolution for the whitecat skin and surprisingly the frame rate didnt change at all. In theory, wouldn't gura skin be faster because of lower resolution? or maybe it is because of the weird resolution it has

**Reply by GopoBoi** [reply]
the 2nd skin looks to have a "fullscreen" hpbar background, so that's definitely not helping.

**Reply by khoibut** [reply]
After messing around for a bit replacing the second skin hp bar with the first skin the frame rates have improved by quite a lot now going at 180 fps average

**Reply by TheFantasticLoki** [reply]
Based on what khoibut says above, is there any way to optimize the rendering of completely transparent pixels? Something like ignoring rendering anything below like 1 or 2 alpha? This kind of performance drop feels very similar to gosumemory in game overlay as it also suffers a massive framerate drop when the render area is fullscreen even if it's only showing a line of text as it's always trying to render pixels that have 0 alpha. 

MANY MANY MANY skins especially very custom/hd skins often have fullscreen hp bars in order to add play area bounds or key overlay backgrounds etc, so I feel that optimizing the way the game renders or handles these types of hp bars should be focused on rather than just saying don't use that hp bar cause a LOT of people have paid, imo, FAR too much money on custom crafted skins and to lose the ability to use your skin you payed sometimes over $300 for just cause of this small issue feels extremely wrong.

**Reply by peppy** [reply]
You should not be using such elements. You should crop them down. While it's probably not the only problem in this thread, there's no way to improve this, which is why you *shouldn't be doing it in the first place*.

Please refrain from commenting further, there is no need for this kind of response.

**Comment by peppy** [comment]
If reporting a skin issue, please start by linking to the skin.

**Reply by khoibut** [reply]
Sorry I couldn't attach it because GitHub didn't allow me to, I added google drive links for both of the skins.
